### PR TITLE
Added Custom App Bar Component

### PR DESCRIPTION
**Pull request by MKamil99**
Implemented custom App Bar with a "Recipes" title and clickable SearchIcon, which is responsible for showing Search Bar. All of the colors are taken from the Figma project.

image
image

### CODE REVIEW

File: src/components/CustomAppBar.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+import { makeStyles } from '@material-ui/core/styles';
+import AppBar from '@material-ui/core/AppBar';
+import Toolbar from '@material-ui/core/Toolbar';
+import Typography from '@material-ui/core/Typography';
+import InputBase from '@material-ui/core/InputBase';
+import IconButton from '@material-ui/core/IconButton';
+import SearchIcon from '@material-ui/icons/Search';
+import CloseIcon from '@material-ui/icons/Close'
+
+
```
Comment (Madoxen) [inline src/components/CustomAppBar.js:16]: Couple things to correct:
- Add callback when the user is submitting the search
- Make app bar header not selectable by user

**Review by Madoxen — CHANGES_REQUESTED** [review]

**Comment by MKamil99** [comment]
New display:

image
image
image

File: src/components/CustomAppBar.js [hunk 2]
```diff
@@ -3,49 +3,82 @@
...
 import { makeStyles } from '@material-ui/core/styles';
 import AppBar from '@material-ui/core/AppBar';
 import Toolbar from '@material-ui/core/Toolbar';
 import Typography from '@material-ui/core/Typography';
 import InputBase from '@material-ui/core/InputBase';
 import IconButton from '@material-ui/core/IconButton';
 import SearchIcon from '@material-ui/icons/Search';
-import CloseIcon from '@material-ui/icons/Close'
+import CloseIcon from '@material-ui/icons/Close';
+import ArrowBackIcon from '@material-ui/icons/ArrowBack';
```
Comment (MellerB) [inline src/components/CustomAppBar.js]: default importów unikamy jak ognia!!!

File: src/components/CustomAppBar.js [hunk 3]
```diff
@@ -3,49 +3,82 @@
...
+
+    // State responsible for displaying "clear button" 
+    // only if there is anything to clear:
+    const [canClear, setCanClear] = useState(false);
+    
+    // Searchbar's value:
+    const searchBarInput = useRef("");
+
+    // Reacting for clicking search button in search bar:
+    const searchButtonEvent = () => props.search(searchBarInput.current.value);
```
Comment (MellerB) [inline src/components/CustomAppBar.js]: nie ma potrzeby komentowania wszystkego jak szalony 
//a teraz zrobię zmienną toster
toster = zróbToster()

File: src/components/CustomAppBar.js (hunk 2 above)
Comment (MellerB) [reply src/components/CustomAppBar.js]: import React, { useState } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import {IconButton, AppBar, Toolbar, Typography, InputBase, IconButton} from '@material-ui/core'
import {SearchIcon, CloseIcon} from '@material-ui/icons'

**Review by Madoxen — APPROVED** [review]

**Comment by Madoxen** [comment]
Fine by me :)

**Pull request merged**